src/db/z_db_pgsql.erl

```diff
@@ -181,9 +182,9 @@ connect(Args, RetryCt) ->
                               [Hostname, Port, ?CONNECT_RETRY_SLEEP div 1000, self()]),
                 timer:sleep(?CONNECT_RETRY_SLEEP),
                 connect(Args, RetryCt+10);
-            {error, {error,fatal,<<"53300">>,_ErrorMsg,_ErrorArgs}} ->
+            {error, #error{ codename = too_many_connections }} ->
                 too_many_connections(Args, RetryCt);
-            {error, <<"53300">>} ->
+            {error, too_many_connections} ->
```

Comment:
Can this one still match?

Reply:
I am actually a  bit unsure.  According to the type specs of the connect not though.

Reply:
Checked again:

```erlang
-type connect_error() ::
        #error{}
      | {unsupported_auth_method, atom()}
      | invalid_authorization_specification
      | invalid_password.
```

Reply:
In that case, let’s remove the redundant patterns.

Reply:
Let's assume the type spec is valid and adapt our code.